model/project_parser.go

```diff
@@ -445,10 +445,13 @@ func LoadProjectForVersion(v *Version, identifier string, shouldSave bool) (*Pro
 	}
 
 	if evergreen.UseParserProject && ppFromDB != nil {
-		// TODO: check here for ppFromDB.ConfigUpdateNumber >= v.ConfigUpdateNumber
-		ppFromDB.Identifier = identifier
-		p, err := TranslateProject(ppFromDB)
-		return p, ppFromDB, err
+		// if parser project config number is old then there was a race,
+		// and we should default to the version config
+		if ppFromDB.ConfigUpdateNumber >= v.ConfigUpdateNumber {
```

Comment:
I might be misunderstanding, but: If the config update numbers are only updated in generate.tasks, then for versions that don't have generators, `ppFromDB.ConfigupdateNumber == v.ConfigUpdateNumber == 0`, and therefore (if `evergreen.UseParserProject == true`) we'll always fall through to using `v.Config`.

Reply:
If the config update numbers are equal, then we DO use parser project (`>=`). We only fall through if the parser project number is less than the version's.